app/javascript/packs/webauthn-authenticate.js

```diff
@@ -1,29 +1,42 @@
 const WebAuthn = require('../app/webauthn');
 
-function webauthn() {
+async function webauthn() {
   // If webauthn is not supported redirect back to the 2fa options list
   const webauthnInProgressContainer = document.getElementById('webauthn-auth-in-progress');
   const webauthnSuccessContainer = document.getElementById('webauthn-auth-successful');
 
-  if (!WebAuthn.isWebAuthnEnabled()) {
+  const webauthnPlatformRequested =
+    webauthnInProgressContainer.dataset.platformAuthenticatorRequested === 'true';
+  const multipleFactorsEnabled =
+    webauthnInProgressContainer.dataset.multipleFactorsEnabled === 'true';
+  const webauthnPlatformEnabled = await window.PublicKeyCredential?.isUserVerifyingPlatformAuthenticatorAvailable();
+
+  if (!WebAuthn.isWebAuthnEnabled() || (webauthnPlatformRequested && !webauthnPlatformEnabled && !multipleFactorsEnabled)) {
     const href = webauthnInProgressContainer.getAttribute('data-webauthn-not-enabled-url');
     window.location.href = href;
   }
```

Comment:
It existed before, but: There should either be a `return;` or an `else` for the remainder of the function, to avoid the rest of the logic from being run. It only worked before because `WebAuthn.verifyWebauthnDevice` would likely throw an error anyways, since we were considering only whether webauthn was enabled. But in the end, the user would still be redirected. With the new logic, it might be that the user will be prompted to present their key, even though we want to redirect them.